jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerEntry.java

```diff
@@ -16,37 +16,92 @@
 
 package com.google.cloud.tools.jib.image;
 
-import com.google.common.collect.ImmutableList;
+import com.google.common.annotations.VisibleForTesting;
 import java.nio.file.Path;
+import java.util.Objects;
 
 /**
  * Represents an entry in the layer. A layer consists of many entries that can be converted into tar
  * archive entries.
  */
 public class LayerEntry {
 
+  private final Path sourceFile;
+  private final Path extractionPath;
+
   /**
-   * The source files to build from. Source files that are directories will have all subfiles in the
-   * directory added (but not the directory itself).
+   * Instantiates with a source file and the path to place the source file in the container file
+   * system.
    *
-   * <p>The source files are specified as a list instead of a set to define the order in which they
-   * are added.
+   * @param sourceFile the source file to add to the layer
+   * @param extractionPath the path to place the source file in the container file system (relative
```

Comment:
Does this mean `extractionPath` shouldn't start with `/`?

Reply:
`extractionPath` can start with `/` or not - it will be normalized when building the actual tarball archive.